### PR TITLE
[CLOUDTRUST-2109] authorizations management

### PR DESCRIPTION
**Pull request by harture**

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
 [[constraint]]
   name = "github.com/cloudtrust/common-service"
-  branch = "master"
+  branch = "CLOUDTRUST-2109_authorizationsManagement"
   #version = "v1.2.1"
```
Comment (bsoniam) [inline Gopkg.toml]: Do you know what comment I have here ? :D

File: groups.go [hunk 2]
```diff
@@ -22,3 +25,38 @@ func (c *Client) GetGroup(accessToken string, realmName string, groupID string)
 	var err = c.get(accessToken, &resp, url.Path(groupByIDPath), url.Param("realm", realmName), url.Param("id", groupID))
 	return resp, err
 }
+
+// CreateGroup creates the group from its GroupRepresentation. The group name must be unique.
+func (c *Client) CreateGroup(accessToken string, reqRealmName string, group GroupRepresentation) (string, error) {
+	return c.post(accessToken, nil, url.Path(groupsPath), url.Param("realm", reqRealmName), body.JSON(group))
+}
+
+// DeleteGroup delete a specific group’s representation
```
Comment (bsoniam) [inline groups.go]: DeleteGroup deletes a specific group representation

File: groups.go [hunk 3]
```diff
@@ -22,3 +25,38 @@ func (c *Client) GetGroup(accessToken string, realmName string, groupID string)
...
+func (c *Client) DeleteGroup(accessToken string, realmName string, groupID string) error {
+	return c.delete(accessToken, url.Path(groupByIDPath), url.Param("realm", realmName), url.Param("id", groupID))
+}
+
+// AssignClientRole assign client roles to a specific group
```
Comment (bsoniam) [inline groups.go]: assigns

File: groups.go [hunk 4]
```diff
@@ -22,3 +25,38 @@ func (c *Client) GetGroup(accessToken string, realmName string, groupID string)
...
+func (c *Client) AssignClientRole(accessToken string, realmName string, groupID string, clientID string, roles []RoleRepresentation) error {
+	_, err := c.post(accessToken, nil, url.Path(groupClientRoleMappingPath), url.Param("realm", realmName), url.Param("id", groupID), url.Param("clientId", clientID), body.JSON(roles))
+	return err
+}
+
+// RemoveClientRole delete client roles to a specific group
```
Comment (bsoniam) [inline groups.go]: deletes

File: groups.go [hunk 5]
```diff
@@ -22,3 +25,38 @@ func (c *Client) GetGroup(accessToken string, realmName string, groupID string)
...
+func (c *Client) RemoveClientRole(accessToken string, realmName string, groupID string, clientID string, roles []RoleRepresentation) error {
+	return c.delete(accessToken, url.Path(groupClientRoleMappingPath), url.Param("realm", realmName), url.Param("id", groupID), url.Param("clientId", clientID), body.JSON(roles))
+}
+
+// GetGroupClientRoles get client roles assigned to a specific group
```
Comment (bsoniam) [inline groups.go]: gets

File: groups.go [hunk 6]
```diff
@@ -22,3 +25,38 @@ func (c *Client) GetGroup(accessToken string, realmName string, groupID string)
...
+func (c *Client) GetGroupClientRoles(accessToken string, realmName string, groupID string, clientID string) ([]RoleRepresentation, error) {
+	var roles = []RoleRepresentation{}
+	var err = c.get(accessToken, &roles, url.Path(groupClientRoleMappingPath), url.Param("realm", realmName), url.Param("id", groupID), url.Param("clientId", clientID))
+	return roles, err
+}
+
+// GetAvailableGroupClientRoles get client roles available to a specific group
```
Comment (bsoniam) [inline groups.go]: gets - you ate too many s's at lunch :)

File: Gopkg.toml (hunk 1 above)
Comment (harture) [reply Gopkg.toml]: I guess ;)
It will be fixed just before merge. It is needed to make it compile

File: groups.go (hunk 6 above)
Comment (harture) [reply groups.go]: I don't know if I ate them or hate them....

Comment (bsoniam) [reply groups.go]: good joke!

**Pull request merged**